bug: java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Request chat/new failed with messa...

### Cody Version

5.5.9

### IDE Information

RustRover 2024.1 EAP
Build #RR-241.15989.180, built on May 7, 2024
Licensed to RustRover EAP User
Expiration date: July 6, 2024
Runtime version: 17.0.10+1-b1207.14 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
GC: G1 Young Generation, G1 Old Generation
Memory: 4050M
Cores: 32
Registry:
  ide.experimental.ui=true
  llm.ai.assistant.toolwindow.activation.on.start=false
  org.toml.json.schema=false
  terminal.new.ui.show.promotion=false
  transferSettings.vscode.onlyCargoToml=true
Non-Bundled Plugins:
  com.sourcegraph.jetbrains (5.5.9)


### Describe the bug

Logged into sg04 w/ usertesting01 account, credentials in 1pass. Chatted to Cody.

### Expected behavior

Chats should work.

### Additional context

Stacktrace:
```text
java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Request chat/new failed with message: No chat model found in server-provided config
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at com.sourcegraph.cody.chat.AgentChatSession.submitMessageToAgent$lambda$1(AgentChatSession.kt:109)
	at com.sourcegraph.cody.agent.CodyAgentService$Companion$withAgent$2$1$task$1.invokeSuspend(CodyAgentService.kt:185)
	at com.sourcegraph.cody.agent.CodyAgentService$Companion$withAgent$2$1$task$1.invoke(CodyAgentService.kt)
	at com.sourcegraph.cody.agent.CodyAgentService$Companion$withAgent$2$1$task$1.invoke(CodyAgentService.kt)
	at com.sourcegraph.cody.agent.CodyAgentService$Companion.coWithAgent(CodyAgentService.kt:241)
	at com.sourcegraph.cody.agent.CodyAgentService$Companion$withAgent$2$1.invokeSuspend(CodyAgentService.kt:190)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at com.sourcegraph.cody.agent.CodyAgentService$Companion.withAgent$lambda$2(CodyAgentService.kt:182)
	at com.intellij.openapi.application.impl.RwLockHolder$executeOnPooledThread$1.run(RwLockHolder.kt:154)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at com.intellij.util.concurrency.ContextCallable.call(ContextCallable.java:32)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Request chat/new failed with message: No chat model found in server-provided config
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:209)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	... 1 more

```
Additional info: ```org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Request chat/new failed with message: No chat model found in server-provided config```

--- a/src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt
+++ b/src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt
@@ -108,7 +108,8 @@ class CodyToolWindowContent(private val project: Project) {
   @RequiresEdt
   fun refreshPanelsVisibility() {
     val codyAuthenticationManager = CodyAuthenticationManager.getInstance(project)
-    if (codyAuthenticationManager.getAccounts().isEmpty()) {
+    if (codyAuthenticationManager.getAccounts().isEmpty() ||
+        codyAuthenticationManager.getActiveAccountTier().getNow(null) == null) {
       allContentLayout.show(allContentPanel, SIGN_IN_PANEL)
       return
     }

--- a/src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt
+++ b/src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt
@@ -115,21 +115,15 @@ private constructor(
                   addEnhancedContext = chatPanel.isEnhancedContextEnabled(),
                   contextFiles = contextItems)
 
-          try {
-            val request =
-                agent.server.chatSubmitMessage(
-                    ChatSubmitMessageParams(connectionId.get().get(), message))
+          val request =
+              agent.server.chatSubmitMessage(
+                  ChatSubmitMessageParams(connectionId.get().get(), message))
 
-            GraphQlLogger.logCodyEvent(project, "chat-question", "submitted")
+          GraphQlLogger.logCodyEvent(project, "chat-question", "submitted")
 
-            createCancellationToken(
-                onCancel = { request.cancel(true) },
-                onFinish = { GraphQlLogger.logCodyEvent(project, "chat-question", "executed") })
-          } catch (e: Exception) {
-            createCancellationToken(onCancel = {}, onFinish = {})
-            handleException(e)
-            throw e
-          }
+          createCancellationToken(
+              onCancel = { request.cancel(true) },
+              onFinish = { GraphQlLogger.logCodyEvent(project, "chat-question", "executed") })
         },
         onFailure = { e ->
           createCancellationToken(onCancel = {}, onFinish = {})

